python/ray/experimental/state/api.py

```diff
@@ -146,6 +146,208 @@ def list(
 
         return response["data"]["result"]
 
+    def get(
+        self,
+        resource: StateResource,
+        id: str,
+        options: Optional[GetApiOptions],
+        _explain: bool = False,
+    ) -> Optional[
+        Union[
+            ActorState,
+            PlacementGroupState,
+            NodeState,
+            WorkerState,
+            TaskState,
+            ObjectState,
+        ]
+    ]:
+        """Get resources states by id
+
+        Args:
+            resource_name: Resource names, i.e. 'workers', 'actors', 'nodes',
+                'placement_groups', 'tasks', 'objects'.
+                'jobs' and 'runtime-envs' are not supported yet.
+            id: ID for the resource, i.e. 'node_id' for nodes.
+            options: List options. See `GetApiOptions` for details.
+            _explain: Print the API information such as API
+                latency or failed query information.
+
+        Returns:
+            None if not found, and found:
+            - ActorState for actors
+            - PlacementGroupState for placement groups
+            - NodeState for nodes
+            - WorkerState for workers
+            - TaskState for tasks
+
+            Empty list for objects if not found, or list of ObjectState for objects
+
+        Raises:
+            This doesn't catch any exceptions raised when the underlying request
+            call raises exceptions. For example, it could raise `requests.Timeout`
+            when timeout occurs.
+
+            ValueError:
+                if the resource could not be GET by id, i.e. jobs and runtime-envs.
+
+        """
+        # TODO(rickyyx): Make GET not using filters on list operation
+        params = self._make_param(options)
+
+        RESOURCE_ID_KEY_NAME = {
+            StateResource.NODES: "node_id",
+            StateResource.ACTORS: "actor_id",
+            StateResource.PLACEMENT_GROUPS: "placement_group_id",
+            StateResource.WORKERS: "worker_id",
+            StateResource.TASKS: "task_id",
+            StateResource.OBJECTS: "object_id",
+        }
+        if resource not in RESOURCE_ID_KEY_NAME:
+            raise ValueError(f"Can't get {resource.name} by id.")
+
+        params["filter_keys"] = [RESOURCE_ID_KEY_NAME[resource]]
```

Comment:
Why don't we just call self.list here?

Reply:
We could - was prepping for the separation of get from list later. A bit of pre-mature factoring here.